### PR TITLE
feat: add new query APIs

### PR DESCRIPTION
**Pull request by Pythonberg1997**
### Description

This pr is to add some new query APIs for group and group member.

### Changes

Notable changes: 
* add new query APIs for group and group member

### CODE REVIEW

File: proto/greenfield/storage/query.proto [hunk 1]
```diff
@@ -128,6 +128,20 @@ service Query {
     option (google.api.http).get = "/greenfield/storage/is_price_changed/{bucket_name}";
   }
 
+  // Queries whether some members are in the group.
+  rpc QueryGroupMembersExist(QueryGroupMembersExistRequest) returns (QueryGroupMembersExistResponse) {
+    option (google.api.http).get = "/greenfield/storage/group_members_exist/{group_id}/{members}";
```
Comment (realuncle) [inline proto/greenfield/storage/query.proto:133]: the path is strange, how `{members}` can be represent in a path

File: proto/greenfield/storage/query.proto [hunk 2]
```diff
@@ -356,4 +370,27 @@ message QueryIsPriceChangedResponse {
...
   ];
 }
+
+message QueryGroupMembersExistRequest {
+  string group_id = 1;
+  repeated string members = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+}
+
+message QueryGroupMembersExistResponse {
+  repeated bool are_members = 1;
```
Comment (realuncle) [inline proto/greenfield/storage/query.proto:380]: I think it is better not only boolean, but response the accordingly accounts as well.

File: proto/greenfield/storage/query.proto [hunk 3]
```diff
@@ -356,4 +370,27 @@ message QueryIsPriceChangedResponse {
...
+}
+
+message QueryGroupsExistRequest {
+  repeated string group_names = 1;
+  repeated string group_owners = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
```
Comment (realuncle) [inline proto/greenfield/storage/query.proto:385]: `group_owners` is not repeated I think

File: proto/greenfield/storage/query.proto [hunk 4]
```diff
@@ -128,6 +128,20 @@ service Query {
...
   }
 
+  // Queries whether some members are in the group.
+  rpc QueryGroupMembersExist(QueryGroupMembersExistRequest) returns (QueryGroupMembersExistResponse) {
+    option (google.api.http).get = "/greenfield/storage/group_members_exist/{group_id}/{members}";
+  }
+
+  // Queries whether some groups are exist.
+  rpc QueryGroupsExist(QueryGroupsExistRequest) returns (QueryGroupsExistResponse) {
+    option (google.api.http).get = "/greenfield/storage/groups_exist/{group_names}/{group_owners}";
```
Comment (Fynnss) [inline proto/greenfield/storage/query.proto:138]: - Query multiple groups under one user at once
- Query multiple groups under multiple users at once

Shall we use a more generic ways to support?

File: proto/greenfield/storage/query.proto [hunk 5]
```diff
@@ -356,4 +370,27 @@ message QueryIsPriceChangedResponse {
...
+message QueryGroupsExistRequest {
+  repeated string group_names = 1;
+  repeated string group_owners = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+}
+
+message QueryGroupsExistByIdRequest {
+  repeated string group_ids = 1;
+}
+
+message QueryGroupsExistResponse {
```
Comment (Fynnss) [inline proto/greenfield/storage/query.proto:392]: Shall we unify the name of the API? E,g. `HeadGroup`, `HeadGroups`, `HeadGroupById`, `HeadGroupsByIds`, `HeadGroupMember`, `HeanGroupMemebers`

Comment (Pythonberg1997) [reply proto/greenfield/storage/query.proto:392]: OK

File: proto/greenfield/storage/query.proto (hunk 4 above)
Comment (Pythonberg1997) [reply proto/greenfield/storage/query.proto:138]: OK. What about nested list? `[groupName1, groupName2, ...]` and `[[group1_members1, ...], [group2_members1, ...], ...]

Comment (Pythonberg1997) [reply proto/greenfield/storage/query.proto:138]: To keep things simple, I think we'd better only support `Query multiple groups under one user at once`.

**Review by guagualvcha — APPROVED** [review]

**Review by Fynnss — APPROVED** [review]

**Pull request merged**